Source/CarthageKit/Xcode.swift

```diff
@@ -1321,6 +1330,13 @@ public func stripFramework(frameworkURL: NSURL, keepingArchitectures: [String],
 		.concat(sign)
 }
 
+/// Strips a dSYM from unexpected architectures.
+public func stripDSYM(dSYMURL: NSURL, keepingArchitectures: [String]) -> SignalProducer<(), CarthageError> {
+	return architecturesInPackage(dSYMURL)
+		.filter { !keepingArchitectures.contains($0) }
+		.flatMap(.Concat) { stripArchitecture(dSYMURL, $0) }
```

Comment:
This this repeats the code at the top of `stripFramework`, can we give it a more generic name and call it from there as well?